src/StorageSync/StorageSync/ServerEndpoint/RemoveServerEndpointCommand.cs

```diff
@@ -180,7 +183,52 @@ public override void ExecuteCmdlet()
                 {
                     if (Force || ShouldContinue(string.Format(StorageSyncResources.RemoveServerEndpointPromptFormat, Target), string.Empty))
                     {
+
+                        StorageSyncModels.ServerEndpoint serverEndpoint = null;
+
+                        try
+                        {
+                            serverEndpoint = StorageSyncClientWrapper.StorageSyncManagementClient.ServerEndpoints.Get(resourceGroupName, storageSyncServiceName, parentResourceName, resourceName);
+                        }
+                        catch(Exception ex)
+                        {
+                            StorageSyncClientWrapper.VerboseLogger.Invoke($"Skipping RoleAssignments removal. Unable to get ServerEndpoint with error {ex.Message}");
+                        }
+
                         StorageSyncClientWrapper.StorageSyncManagementClient.ServerEndpoints.Delete(resourceGroupName, storageSyncServiceName, parentResourceName, resourceName);
+
+                        if (serverEndpoint != null)
+                        {
+                            // Get Registered Server
+                            var serverResourceIdentifier = new ResourceIdentifier(serverEndpoint.ServerResourceId);
+                            StorageSyncModels.RegisteredServer registeredServer = StorageSyncClientWrapper.StorageSyncManagementClient.RegisteredServers.Get(resourceGroupName, storageSyncServiceName, serverResourceIdentifier.ResourceName);
+
+                            if (registeredServer == null)
+                            {
+                                throw new PSArgumentException(StorageSyncResources.MissingServerResourceIdErrorMessage);
+                            }
+
+                            if (!String.IsNullOrEmpty(registeredServer.ApplicationId) && Guid.TryParse(registeredServer.ApplicationId, out Guid serverIdentityGuid))
+                            {
+                                StorageSyncModels.CloudEndpoint cloudEndpoint = StorageSyncClientWrapper.StorageSyncManagementClient.CloudEndpoints
+                                .ListBySyncGroup(resourceGroupName, storageSyncServiceName, parentResourceName).FirstOrDefault();
+
+                                if (cloudEndpoint == null)
+                                {
+                                    throw new PSArgumentException(StorageSyncResources.MissingParentResourceIdErrorMessage);
+                                }
+                                var storageAccountResourceIdentifier = new ResourceIdentifier(cloudEndpoint.StorageAccountResourceId);
+                                var scope = $"{cloudEndpoint.StorageAccountResourceId}/fileServices/default/fileshares/{cloudEndpoint.AzureFileShareName}";
+                                bool success = StorageSyncClientWrapper.DeleteRoleAssignmentWithIdentity(storageAccountResourceIdentifier.Subscription,
```

Comment:
If Delete SEP, and it is MI enabled topology, remove RBAC to that CEP/FileShare for the server.

Wouldn't we need to do the same cluster logic like for Create (Delete RBAC for all node servers?)

Also is it possible for this RBAC to not exist?

Reply:
We don't fail the SEP command if RBAC deletion fails